dev-packages/application-manager/src/generator/frontend-generator.ts

```diff
@@ -105,18 +117,45 @@ const { join } = require('path');
 const { isMaster } = require('cluster');
 const { fork } = require('child_process');
 const { app, BrowserWindow, ipcMain } = require('electron');
+const EventEmitter = require('events');
+const fileSchemeTester = /^file:/;
 
+const localUriEvent = new EventEmitter();
+let localUri = undefined;
 const windows = [];
 
+function setLocalUri(uri) {
+    localUriEvent.emit('update', localUri = uri);
+}
+function resolveLocalUriFromPort(port) {
+    setLocalUri('file://' + join(__dirname, '../../lib/index.html') + '?port=' + port);
+}
+
 function createNewWindow(theUrl) {
-    const newWindow = new BrowserWindow({ width: 1024, height: 728, show: !!theUrl });
+    const config = {
+        width: 1024,
+        height: 728,
+        show: !!theUrl
+    };
+
+    // Converts 'localhost' to the running local backend endpoint
+    if (localUri && theUrl === 'localhost') {
+        theUrl = localUri;
+    }
+
+    if (!!theUrl && !fileSchemeTester.test(theUrl)) {
+        config.webPreferences = {
+            // nodeIntegration: false,
```

Comment:
This is here so that we can easily enable whatever option is required.

I was hoping to have it working in this PR but turning off the nodeIntegration makes Electron fail.

As long as it is here, people will be able to see it more easily and make the required changes later, when needed.